ledger-tool/src/bigtable.rs

```diff
@@ -53,6 +53,16 @@ async fn upload(
     };
     let blockstore = Arc::new(blockstore);
 
+    let starting_slot = if let Some(starting_slot) = starting_slot {
+        Ok(starting_slot)
+    } else {
+        // It is possible that the slot returned below could get purged by
+        // LedgerCleanupService before upload_confirmed_blocks() receives the
+        // value. This is ok because upload_confirmed_blocks() doesn't need
+        // the exact slot to be in ledger, the slot is only used as a bound.
+        blockstore.get_first_available_block()
+    };
+    let mut starting_slot = starting_slot?;
```

Comment:
```suggestion
    let starting_slot = match starting_slot {
        Some(slot) => slot,
        None => blockstore.get_first_available_block()?,
    };
```
How about this?

Reply:
(There's also this silly alternative:
```
starting_slot.ok_or_else(||"some temporary err".to_string()).or_else(|_| get_first_available_block(bar2))?;
```
But I prefer the match)

Reply:
Ha yeah, appreciate the creativity on the second one, but the match seems like the clear winner.

On a similar vein - thoughts on keeping the comment? It is more or less covering the thought progression you had on my other PR about whether this introduced a race condition. I'm a little on the fence about whether it is overkill

Reply:
Let's keep it. Could save someone some digging in the future.